sources/mongo/mongo.go

```diff
@@ -0,0 +1,64 @@
+package mongo
+
+/*
+ Package mongo adapts a mgo iterator (or anything that implements
+ the interface) into a optimus.Table for consumption
+
+ example:
+   // Connect to mongo, get your collection then do the following:
+	 iter := collection.Find(nil).Limit(100).Iter()
+	 table := mongo.New(iter)
+*/
+
+import (
+	"gopkg.in/Clever/optimus.v3"
+)
+
+// Iter simulates the mgo.Iter interface so we can remain independent
+type Iter interface {
+	Next(result interface{}) bool
+	Err() error
+}
+
+// New returns a new Table that iterates over all the results of a mongo query.
+func New(iter Iter) optimus.Table {
+	s := &mongoSource{rows: make(chan optimus.Row)}
+	go s.start(iter)
+	return s
+}
+
+// mongoSource type matches the
+type mongoSource struct {
+	err     error
+	rows    chan optimus.Row
+	stopped bool
+}
+
+// start begins feeding rows into the rows channel
+func (s *mongoSource) start(iter Iter) {
+	defer s.Stop()
+	defer close(s.rows)
+	for !s.stopped {
+		r := optimus.Row{}
+		if !iter.Next(&r) {
+			break
+		}
+		s.rows <- r
+	}
+	s.err = iter.Err()
+}
+
+// Rows returns the read side of the channel of optimus Rows from this mongo source
```

Comment:
You don't need to document these since they're private, but if you want to, a common pattern in Go (to avoid duplicating documentation) is to say something like "Rows implements the Table interface", "Err implements the Table interface", etc., basically telling people that the description of that method is in the interface documentation.

I usually don't say anything about these private methods since it's pretty straightforward.